src/hotel_reservations/mlflow_processor.py

```diff
@@ -67,16 +86,44 @@ def __init__(self, model):
 
             def predict(self, context, model_input):
                 if isinstance(model_input, pd.DataFrame):
-                    predictions = self.model.predict(X_test)
+                    predictions = self.model.predict(model_input)
                     predictions = {"Prediction": adjust_predictions(predictions[0])}
                     return predictions
                 else:
                     raise ValueError("Input must be a pandas DataFrame.")
 
-        self.wrapped_model = HousePriceModelWrapper(self.model)
-        self.example_prediction = self.wrapped_model.predict(context=None, model_input=X_test.iloc[0:1])
+        wrapped_model = HousePriceModelWrapper(self.model)
+        example_prediction = wrapped_model.predict(context=None, model_input=X_test.iloc[0:1])
+
+        return wrapped_model, example_prediction
+
+    def model_wrapper_ab_test(self, models, X_test):
+        class HousePriceModelWrapper(mlflow.pyfunc.PythonModel):
```

Comment:
again, would be nicer to have it somewhere else. and confusing to have the same name of the class